File: core/src/com/mygdx/game/model/pickup/PistolPickup.kt
package com.mygdx.game.model.pickup

import com.badlogic.gdx.graphics.Texture
import com.mygdx.game.model.pickup.Pickup
import com.mygdx.game.settings.PISTOL_SPRITE_HEIGHT
import com.mygdx.game.settings.PISTOL_SPRITE_WIDTH

class PistolPickup(
        x: Float = 0f,
        y: Float = 0f,
        texture: Texture) : Pickup(
        x,
        y,
        PISTOL_SPRITE_WIDTH,
        PISTOL_SPRITE_HEIGHT,
        texture
)

File: server/models/pickups/PistolPickup.js
const Pickup = require('./Pickup');
const constants = require('../../settings/constants');
const ProjectileType = require('../projectiles/ProjectileType');

class PistolPickup extends Pickup {
    constructor(x, y, id, ammunition = constants.PISTOL_BULLETS_IN_CHAMBER) {
        super(x, y, constants.PISTOL_SPRITE_WIDTH, constants.PISTOL_SPRITE_HEIGHT, ProjectileType.PISTOL, id,
            ammunition)
    }
}

module.exports = PistolPickup;